src/backend/distributed/commands/cascade_table_operation_for_connected_relations.c

```diff
@@ -68,6 +68,13 @@ CascadeOperationForConnectedRelations(Oid relationId, LOCKMODE lockMode,
 	InvalidateForeignKeyGraph();
 
 	List *fKeyConnectedRelationIdList = GetForeignKeyConnectedRelationIdList(relationId);
+
+	/* early exit if there are no connected relations */
+	if (fKeyConnectedRelationIdList == NIL)
+	{
+		return;
+	}
```

Comment:
very minor comment on style + I think it's much safer to use list_length to check empty lists

```suggestion
	List *fKeyConnectedRelationIdList = GetForeignKeyConnectedRelationIdList(relationId);
	if (list_length(fKeyConnectedRelationIdList) == 0)
	{
		/* early exit if there are no connected relations */
		return;
	}
```